packages/tizen_rpc_port/lib/src/rpc_port_method_channel.dart

```diff
@@ -0,0 +1,200 @@
+import 'dart:typed_data';
+import 'package:flutter/services.dart';
+import 'package:tizen_log/tizen_log.dart';
+
+import 'port.dart';
+import 'proxy_base.dart';
+
+const String _logTag = 'RpcPortMethodChannel';
+
+/// An implementation of [RpcPortPlatform] that uses method channels.
+class MethodChannelRpcPort {
+  static final MethodChannelRpcPort instance = MethodChannelRpcPort();
+
+  /// The method channel used to interact with the native platform.
+  final MethodChannel _channel = const MethodChannel('tizen/rpc_port');
+
+  final Map<String, Stream<dynamic>> _streams = <String, Stream<dynamic>>{};
+
+  String _createKey(String appid, String portName) => '$appid/$portName';
+
+  Future<void> stubCreate(String portName) async {
+    final Map<String, String> args = <String, String>{'portName': portName};
+    return await _channel.invokeMethod('stubCreate', args);
+  }
+
+  Future<void> stubDestroy(String portName) async {
+    final Map<String, String> args = <String, String>{'portName': portName};
+    return await _channel.invokeMethod('stubDestroy', args);
+  }
+
+  Future<Stream<dynamic>> proxyConnect(ProxyBase proxy) async {
+    final String key = _createKey(proxy.appid, proxy.portName);
+    if (_streams.containsKey(key)) {
+      return _streams[key]!;
+    }
+
+    final EventChannel eventChannel = EventChannel('tizen/rpc_port_proxy/$key');
+    _streams[key] = eventChannel.receiveBroadcastStream();
+    final Map<String, dynamic> args = <String, dynamic>{
+      'appid': proxy.appid,
+      'portName': proxy.portName,
+    };
+
+    await _channel.invokeMethod<dynamic>('proxyConnect', args);
+    return _streams[key]!;
+  }
+
+  Future<Stream<dynamic>> proxyConnectSync(ProxyBase proxy) async {
+    final String key = _createKey(proxy.appid, proxy.portName);
+    if (_streams.containsKey(key)) {
+      return _streams[key]!;
+    }
+
+    final EventChannel eventChannel = EventChannel('tizen/rpc_port_proxy/$key');
+    _streams[key] = eventChannel.receiveBroadcastStream();
+    final Map<String, dynamic> args = <String, dynamic>{
+      'appid': proxy.appid,
+      'portName': proxy.portName,
+    };
+
+    await _channel.invokeMethod<dynamic>('proxyConnectSync', args);
+    return _streams[key]!;
+  }
```

Comment:
`proxyConnectSync` should directly call ``RpcPortProxy::ConnectSync` in C++ code. Currently, it returns a stream that calls those functions when stream in listened to.

```suggestion
  Future<void> proxyConnectSync(ProxyBase proxy) async {
    final Map<String, dynamic> args = <String, dynamic>{
      'appid': proxy.appid,
      'portName': proxy.portName,
    };

    await _channel.invokeMethod<dynamic>('proxyConnectSync', args);
  }
```

Same with `proxyConnect`.

Reply:
Your suggestion is nice!
But it will take some time to apply it.
I referred message_port plugin. In a very similar way.
Since it is an internal implementation, I will prepare the patch that applied your opinion later.
Thank you for your feedback.